### PR TITLE
[STACK-1910]: Create nat-pool table in database.

### PR DESCRIPTION
**Pull request by rahulkatre-a10**
## Description
Added migration revision to add nat_pool table along with its model and data-model.

## Jira Ticket
https://a10networks.atlassian.net/browse/STACK-1910

## Technical Approach
- Recognized the column need for the nat_pool table.
- Added model class with attributes mapped to nat_pool table.
- Autogenerated the migration revision from model changes.
- Added data-model for the NATPool model.

## Manual Testing
- Applied the migration by alembic upgrade on Octavia database, should create nat_pool table.
- Reverted the migration by alembic downgrade on Octavia DB, should drop the nat_pool table.
- Checked the properties of the nat_pool table after applying migration.

### CODE REVIEW

File: a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    op.create_table('nat_pool',
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('name', sa.String(length=1024), nullable=False),
+    sa.Column('subnet_id', sa.String(length=36), nullable=False),
+    sa.Column('start_ip_address', sa.String(length=64), nullable=False),
+    sa.Column('end_ip_address', sa.String(length=64), nullable=False),
+    sa.Column('member_ref_count', sa.Integer(), nullable=True),
+    sa.PrimaryKeyConstraint('id')
+    )
+    # ### end Alembic commands ###
```
Comment (afrin-chakure-a10) [inline a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py:30]: Please remove this

File: a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+# revision identifiers, used by Alembic.
+revision = 'a41260fb4ccd'
+down_revision = '05b1446c7f20'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
```
Comment (afrin-chakure-a10) [inline a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py:20]: Please remove this

File: a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    sa.Column('end_ip_address', sa.String(length=64), nullable=False),
+    sa.Column('member_ref_count', sa.Integer(), nullable=True),
+    sa.PrimaryKeyConstraint('id')
+    )
+    # ### end Alembic commands ###
+
+
+def downgrade():
+    op.drop_table('nat_pool')
+    # ### end Alembic commands ###
```
Comment (afrin-chakure-a10) [inline a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py:35]: Please remove this

**Review by afrin-chakure-a10 — COMMENTED** [review]
minor comments

File: a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py (hunk 1 above)
Comment (rahulkatre-a10) [reply a10_octavia/db/migration/alembic_migrations/versions/3f576e3cd730_added_nat_pool_table.py]: removed the comments

**Review by NehaKembalkarA10 — APPROVED** [review]

**Comment by rahulkatre-a10** [comment]
> can we extend the length to 64?

changed to 64

**Review by ytsai-a10 — APPROVED** [review]

**Pull request merged**